script/test

```diff
@@ -127,6 +127,7 @@ build_simulation()
 
     if [[ ${version} != "1.1" ]]; then
         options+=("-DOT_CSL_RECEIVER=ON")
+        options+=("-DOT_CHANNEL_MANAGER_CSL=ON")
```

Comment:
Ok, this breaks the matrix build jobs. Do we want to add a separate job just for this? As other optional options, eg. `ChannelManager` and `ChannelMonitor` are also enabled in the test build, it might be acceptable we keep it in `script/test` ?

Reply:
I checked the failed runs, one example is: https://github.com/openthread/openthread/actions/runs/7303232394/job/19903342538#step:7:16247

I mean you can add an option in the environment of the CI config(`.github/workflows/simulation-1.2.yml`):
```
  thread-1-3:                                                                                                                                                                                                                                                                             
    name: thread-1-3-${{ matrix.compiler.c }}-${{ matrix.arch }}                                                                                                                                                                                                                          
    runs-on: ubuntu-20.04                                                                                                                                                                                                                                                                 
    env:                                                                                                                                                                                                                                                                                  
      CFLAGS: -${{ matrix.arch }}                                                                                                                                                                                                                                                         
      CXXFLAGS: -${{ matrix.arch }}                                                                                                                                                                                                                                                       
      LDFLAGS: -${{ matrix.arch }}                                                                                                                                                                                                                                                        
      COVERAGE: 1                                                                                                                                                                                                                                                                         
      THREAD_VERSION: 1.3                                                                                                                                                                                                                                                                 
      VIRTUAL_TIME: 1                                                                                                                                                                                                                                                                     
      INTER_OP: 1                                                                                                                                                                                                                                                                         
      INTER_OP_BBR: 1
      EXPERIMENTAL_FEATURES: 1      <== Add this one                                                                                                                                                                                                                                                                     
      CC: ${{ matrix.compiler.c }}                                                                                                                                                                                                                                                        
      CXX: ${{ matrix.compiler.cxx }}     
```
Then in the `script/test`, convert the option into cmake options:
```
EXPERIMENTAL_FEATURES="${EXPERIMENTAL_FEATURES:-0}"       <== Add this                                                                                                                                                                                                                                              
readonly EXPERIMENTAL_FEATURES                                                      <== Read it from environment 

build_simulation()                                                                                                                                                                                                                                                                        
{                                                                                                                                                                                                                                                                                         
    local version="$1"                                                                                                                                                                                                                                                                    
    local options=(                                                                                                                                                                                                                                                                       
        "-DBUILD_TESTING=ON"                                                                                                                                                                                                                                                              
        "-DOT_ANYCAST_LOCATOR=ON"                                                                                                                                                                                                                                                         
        "-DOT_DNS_CLIENT=ON"                                                                                                                                                                                                                                                              
        "-DOT_DNS_DSO=ON"                                                                                                                                                                                                                                                                 
        "-DOT_DNSSD_SERVER=ON"                                                                                                                                                                                                                                                            
        ......                                                                                                                                                                                                                                                
    )                                                                                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                                                          
    if [[ ${FULL_LOGS} == 1 ]]; then                                                                                                                                                                                                                                                      
        options+=("-DOT_FULL_LOGS=ON")                                                                                                                                                                                                                                                    
    fi

    ......

    if [[ ${EXPERIMENTAL_FEATURES} == 1 ]]; then                                                                                                                                                                                                                                                      
        options+=("-DOT_CHANNEL_MANAGER_CSL=ON")                                                                                                                                                                                                                                                    
    fi                                                                     
```

What I think is that in this utility script (`script/test`), all the options enabled by default are standard options (like CSL, Link Metrics). So I would prefer not making non-standard features as default options in the script. Maybe there are some non-standard options being turned on by default. I think they are not turned on in the `script/test`, but turned on elsewhere (in `/etc/options.cmake` or the core config file) for some reasons.

Reply:
Thx, changed accordingly.